rocky/onboarding/views.py

```diff
@@ -100,237 +92,74 @@ class OnboardingChooseReportTypeView(
-
-    def get_context_data(self, **kwargs: Any) -> dict[str, Any]:
-        context = super().get_context_data(**kwargs)
-        context["select_boefjes_form"] = self.get_form()
-        return context
-
-
 class OnboardingSetupScanOOIInfoView(
     OrganizationPermissionRequiredMixin,
     KatIntroductionStepsMixin,
     OrganizationView,
     TemplateView,
 ):
-    template_name = "step_3a_setup_scan_ooi_info.html"
-    current_step = 3
-    permission_required = "tools.can_scan_organization"
-
-
-class OnboardingOOIForm(OOIForm):
     """
-    hidden_fields - key (field name) value (field value) pair that will rendered as hidden field
+    4. Explanation that an object is needed to make scans.
     """
 
-    def __init__(
-        self, hidden_fields: dict[str, str], ooi_class: type[OOI], connector: OctopoesAPIConnector, *args, **kwargs
-    ):
-        self.hidden_ooi_fields = hidden_fields
-        super().__init__(ooi_class, connector, *args, **kwargs)
-
-    def get_fields(self):
-        return self.generate_form_fields(self.hidden_ooi_fields)
+    template_name = "step_3a_setup_scan_ooi_info.html"
+    current_step = 3
+    permission_required = "tools.can_scan_organization"
 
 
 class OnboardingSetupScanOOIAddView(
     OrganizationPermissionRequiredMixin,
     KatIntroductionStepsMixin,
-    BaseOOIFormView,
+    SingleOOITreeMixin,
+    FormView,
 ):
+    """
+    5. The user will create a URL object. Shows a form and validation to create object.
+    """
+
     template_name = "step_3b_setup_scan_ooi_add.html"
     current_step = 3
     permission_required = "tools.can_scan_organization"
-    form_class = OnboardingOOIForm
-    hidden_form_fields = {
-        "network": {
-            "ooi": Network(name="internet"),
-        }
-    }
+    form_class = OnboardingCreateObjectURLForm
 
     def setup(self, request, *args, **kwargs):
         super().setup(request, *args, **kwargs)
-        self.ooi_class = self.get_ooi_class()
-
-    def get_hidden_form_fields(self):
-        hidden_fields = {}
-        bytes_client = get_bytes_client(self.organization.code)
-
-        for field_name, params in self.hidden_form_fields.items():
-            ooi, created = get_or_create_ooi(self.octopoes_api_connector, bytes_client, params["ooi"], self.observed_at)
-            hidden_fields[field_name] = ooi.primary_key
-
-            if created:
-                messages.success(
-                    self.request,
-                    _(
-                        "OpenKAT added the following required object to your object list to complete your request: {}"
-                    ).format(str(ooi)),
-                )
-        return hidden_fields
-
-    def get_form_kwargs(self):
-        kwargs = super().get_form_kwargs()
-        hidden_fields = self.get_hidden_form_fields()
-        kwargs.update({"hidden_fields": hidden_fields, "initial": hidden_fields})
+        self.report_type = self.request.GET.get("report_type", "")
```

Comment:
I thought the onboarding always used the DNS report?

Reply:
Yes, the idea here is to use more report types in the future. But that will follow in another PR if necessary.